client/knode/validator/validator.go

```diff
@@ -39,20 +39,24 @@ type Backend interface {
 }
 
 type Validator interface {
+	ValidatorSafe
 	Start(walletAccount accounts.WalletAccount, deposit *big.Int)
 	Stop() error
 	SetExtra(extra []byte) error
-	Validating() bool
-	Running() bool
 	SetCoinbase(walletAccount accounts.WalletAccount) error
 	SetDeposit(deposit *big.Int) error
 	Pending() (*types.Block, *state.StateDB)
 	PendingBlock() *types.Block
+	Deposits(address *common.Address) ([]*types.Deposit, error)
+	RedeemDeposits() error
+}
+
+type ValidatorSafe interface {
```

Comment:
sorry but this name helps nothing understating what it does. lets try a few?

maybe:

Validator -> Manager/Service
ValidatorSafe -> Validator

more suggestions?

Reply:
I like this naming. How do you think, maybe the opposite naming is better:
Validator -> Validator
ValidatorSafe -> Service ?